sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -288,6 +295,12 @@ private[sql] class JDBCRDD(
     case GreaterThanOrEqual(attr, value) => s"$attr >= ${compileValue(value)}"
     case IsNull(attr) => s"$attr IS NULL"
     case IsNotNull(attr) => s"$attr IS NOT NULL"
+    case In(attr, value) => s"$attr IN (${compileValue(value)})"
+    case Not(In(attr, value)) => s"$attr NOT IN (${compileValue(value)})"
```

Comment:
ISTM we do not need this entry by calling `compileFilter` recursively.
Basically, we should take `ParquetFilters#createFilter` as an example implementation.